src/main/java/org/opensearch/flowframework/util/ParseUtils.java

```diff
@@ -472,4 +473,29 @@ public static Map<String, String> convertStringToObjectMapToStringToStringMap(Ma
             return stringToStringMap;
         }
     }
+
+    /**
+     * Checks if the inputs map contains the specified key and parses the associated value to a generic class.
+     *
+     * @param <T> the type to which the value should be parsed
+     * @param inputs the map containing the input data
+     * @param key the key to check in the map
+     * @param type the class to parse the value to
+     * @throws IllegalArgumentException if the type is not supported
+     * @return the generic type value associated with the key if present, or null if the key is not found
+     */
+    public static <T> T parseIfExists(Map<String, Object> inputs, String key, Class<T> type) {
+        if (!inputs.containsKey(key)) {
+            return null;
+        }
+
+        Object value = inputs.get(key);
+        if (type == Boolean.class) {
+            return type.cast(Booleans.parseBoolean(value.toString()));
+        } else if (type == Float.class) {
+            return type.cast(Float.parseFloat(value.toString()));
+        } else {
+            throw new IllegalArgumentException("Unsupported type: " + type);
```

Comment:
This is fine for now, but I'd prefer an `UnsupportedOperationException` the next time we touch this code.